src/criteria_assessment/site_identification.jl

```diff
@@ -0,0 +1,262 @@
+"""Methods for identifying potential deployment locations."""
+
+using FLoops, ThreadsX
+
+"""
+    proportion_suitable(subsection::BitMatrix, window::Tuple=(-4,5))::Matrix{Int16}
+
+Calculate the the proportion of the subsection that is suitable for deployments.
+Subsection is the surrounding a rough hectare area centred on each cell of a raster marked
+as being suitable according to user-selected criteria.
+
+# Arguments
+- `x` : Matrix of boolean pixels after filtering with user criteria.
+- `window` : Window size to assess. Default window (-4,5) assesses a square hectare around each target pixel where the resolution of pixels is 10m.
+"""
+function proportion_suitable(x::BitMatrix; window::Tuple=(-4, 5))::Matrix{Int16}
+    x′ = zeros(Int16, size(x))
+
+    @floop for row_col in ThreadsX.findall(x)
+        (row, col) = Tuple(row_col)
+        x_left = max(col + window[1], 1)
+        x_right = min(col + window[2], size(x, 1))
+
+        y_top = max(row + window[1], 1)
+        y_bottom = min(row + window[2], size(x, 2))
+
+        x′[row, col] = Int16(sum(@views x[y_top:y_bottom, x_left:x_right]))
+    end
+
+    return x′
+end
+
+"""
+    filter_distances(
+        target_rast::Raster,
+        gdf::DataFrame,
+        dist_nm
+    )::Raster
+
+Exclude pixels in `target_rast` that are beyond `dist_nm` (nautical miles) from a geometry
+in `gdf`.  `target_rast` and `gdf` should be in the same CRS (EPSG:7844 / GDA2020 for GBR-reef-guidance-assessment).
+
+# Arguments
+- `target_rast` : Raster of suitable pixels (Bool) to filter pixels from.
+- `gdf` : DataFrame with `geometry` column that contains vector objects of interest.
+- `dist_nm` : Filtering distance from geometry object in nautical miles.
+
+# Returns
+- `tmp_areas` : Raster of filtered pixels containing only pixels within target distance
+from a geometry centroid.
+"""
+function filter_distances(
+    target_rast::Raster, gdf::DataFrame, dist; units::String="NM"
+)::Raster
+    tmp_areas = copy(target_rast)
+
+    # First dimension is the rows (latitude)
+    # Second dimension is the cols (longitude)
+    raster_lat = Vector{Float64}(tmp_areas.dims[1].val)
+    raster_lon = Vector{Float64}(tmp_areas.dims[2].val)
+
+    @floop for row_col in findall(tmp_areas)
+        (lat_ind, lon_ind) = Tuple(row_col)
+        point = AG.createpoint()
+
+        lon = raster_lon[lon_ind]
+        lat = raster_lat[lat_ind]
+        AG.addpoint!(point, lon, lat)
+
+        pixel_dists = AG.distance.([point], port_locs.geometry)
+        geom_point = gdf[argmin(pixel_dists), :geometry]
+        geom_point = (AG.getx(geom_point, 0), AG.gety(geom_point, 0))
+
+        dist_nearest = Distances.haversine(geom_point, (lon, lat))
+
+        # Convert from meters to nautical miles
+        if units == "NM"
+            dist_nearest = dist_nearest / 1852
+        end
+
+        # Convert from meters to kilometers
+        if units == "km"
```

Comment:
nit (nit-pick), but could be else if. Also hardcoding the values isn't ideal. Is there some kind of unit conversion lib in Julia? something like `unit_convert(dist_nearest, "m", "NM")`

Reply:
well, better would be returning a function. `unit_converter("m", "NM")(dist_nearest)`
Unitful?
https://painterqubits.github.io/Unitful.jl/stable/conversion/

Reply:
not for this PR, this was more of a general should we use Unitful in our projects question.